### PR TITLE
Ff113 RTCRtpSender/RTCRtpRecieve .getCapabilities() - fixup to not use dictionary

### PR DESCRIPTION
**Pull request by hamishwillee**
`RTCRtpSender.getCapabilities()` and `RTCRtpReciever.getCapabilities()`  return WebRTC codec and header capabilities. These were previously documented as separate dicitionaries - capabilities, that referenced codecs. Neither dictionary had BCD.

This removes the dictionaries and puts the content into the methods. Redirects to `RTCRtpSender.getCapabilities()`.
Also adds a live example so you can see what the data might look like - a bit more "real".

There is some duplication I am not that happy about, but overall much better.

This is part of tidy up fixes for #26146

### CODE REVIEW

File: files/en-us/web/api/rtcrtpreceiver/getcapabilities/index.md [hunk 1]
```diff
@@ -8,13 +8,9 @@ browser-compat: api.RTCRtpReceiver.getCapabilities
 
 {{DefaultAPISidebar("WebRTC")}}
 
-The _static function_
-**`RTCRtpReceiver.getCapabilities()`** returns
-an {{domxref("RTCRtpCapabilities")}} object describing the codecs and capabilities
-supported by {{domxref("RTCRtpReceiver")}}s on the current device.
+The _static function_ **`RTCRtpReceiver.getCapabilities()`** returns an object describing the codec and header extension capabilities supported by {{domxref("RTCRtpReceiver")}} objects on the current device.
```
Comment (teoli2003) [inline files/en-us/web/api/rtcrtpreceiver/getcapabilities/index.md:11]:
```suggestion
The _static method **`RTCRtpReceiver.getCapabilities()`** returns an object describing the codec and header extension capabilities supported by {{domxref("RTCRtpReceiver")}} objects on the current device.
```

File: files/en-us/web/api/rtcrtpreceiver/getcapabilities/index.md [hunk 2]
```diff
@@ -25,16 +21,47 @@ RTCRtpReceiver.getCapabilities(kind)
...
+  - : A string indicating the type of media for which the browser's receiver capabilities are requested.
+    The supported media kinds are: `audio` and `video`.
 
 ### Return value
 
-An {{domxref("RTCRtpCapabilities")}} object stating what capabilities the browser has
-for receiving the specified media kind over an {{domxref("RTCPeerConnection")}}. If the
-browser doesn't have any support for the given media kind, the returned value is
-`null`.
+An object that indicates what capabilities the browser has for receiving the specified media kind over an {{domxref("RTCPeerConnection")}}.
```
Comment (teoli2003) [inline files/en-us/web/api/rtcrtpreceiver/getcapabilities/index.md:29]: A new object is created for each call:
```suggestion
A new object that indicates what capabilities the browser has for receiving the specified media kind over an {{domxref("RTCPeerConnection")}}.
```

File: files/en-us/web/api/rtcrtpreceiver/getcapabilities/index.md [hunk 3]
```diff
@@ -81,6 +144,59 @@ function canReceiveH264() {
...
+    log.textContent += `   channels: ${codec.channels}\n`; //max channels - e.g. 2 is stereo
+    log.textContent += `   clockRate: ${codec.clockRate}\n`; // clock rate in Hz
+    log.textContent += `   sdpFmtpLine: ${codec.sdpFmtpLine}\n`; // mime meda type and subtype
+  });
+}
+```
+
+#### Result
+
+{{ EmbedLiveSample('Get all capabilities', '100%', '500px') }}
```
Comment (teoli2003) [inline files/en-us/web/api/rtcrtpreceiver/getcapabilities/index.md:198]:
```suggestion
{{ EmbedLiveSample('Getting all capabilities', '100%', '500px') }}
```

File: files/en-us/web/api/rtcrtpsender/getcapabilities/index.md [hunk 4]
```diff
@@ -8,13 +8,9 @@ browser-compat: api.RTCRtpSender.getCapabilities
 
 {{DefaultAPISidebar("WebRTC")}}
 
-The _static function_
-**`RTCRtpSender.getCapabilities()`** returns
-an {{domxref("RTCRtpCapabilities")}} object describing the codecs and capabilities
-supported by the {{domxref("RTCRtpSender")}}.
+The _static function_ **`RTCRtpSender.getCapabilities()`** returns an object describing the codec and header extension capabilities supported by the {{domxref("RTCRtpSender")}}.
```
Comment (teoli2003) [inline files/en-us/web/api/rtcrtpsender/getcapabilities/index.md:11]:
```suggestion
The _static method_ **`RTCRtpSender.getCapabilities()`** returns an object describing the codec and header extension capabilities supported by the {{domxref("RTCRtpSender")}}.
```

File: files/en-us/web/api/rtcrtpsender/getcapabilities/index.md [hunk 5]
```diff
@@ -25,16 +21,47 @@ RTCRtpSender.getCapabilities(kind)
...
+  - : A string indicating the type of media for which the browser's send capabilities are requested.
+    The supported media kinds are: `audio` and `video`.
 
 ### Return value
 
-An {{domxref("RTCRtpCapabilities")}} object stating what capabilities the browser has
-for sending the specified media kind over an {{domxref("RTCPeerConnection")}}. If the
-browser doesn't have any support for the given media kind, the returned value is
-`null`.
+An object that indicates what capabilities the browser has for sending the specified media kind over an {{domxref("RTCPeerConnection")}}.
```
Comment (teoli2003) [inline files/en-us/web/api/rtcrtpsender/getcapabilities/index.md:29]:
```suggestion
A new object that indicates what capabilities the browser has for sending the specified media kind over an {{domxref("RTCPeerConnection")}}.
```

File: files/en-us/web/api/rtcrtpsender/getcapabilities/index.md [hunk 6]
```diff
@@ -44,29 +71,67 @@ As a static function, this is always called using the form:
...
+```js
+log.textContent = `RTCRtpSender.getCapabilities() supported: ${Object.hasOwn(
+  RTCRtpSender,
+  "getCapabilities"
+)}`;
+```
+
+{{ EmbedLiveSample('Feature support', '100%', '30px') }}
+
+### Check support for a particular codec
```
Comment (teoli2003) [inline files/en-us/web/api/rtcrtpsender/getcapabilities/index.md:129]:
```suggestion
### Checking support for a particular codec
```

File: files/en-us/web/api/rtcrtpsender/getcapabilities/index.md [hunk 7]
```diff
@@ -81,6 +146,55 @@ function canSendH264() {
 }
 ```
 
+### Get all capabilities
```
Comment (teoli2003) [inline files/en-us/web/api/rtcrtpsender/getcapabilities/index.md:149]:
```suggestion
### Getting all capabilities
```

File: files/en-us/web/api/rtcrtpsender/getcapabilities/index.md [hunk 8]
```diff
@@ -81,6 +146,55 @@ function canSendH264() {
...
+  const capabilities = RTCRtpSender.getCapabilities(`${kind}`);
+  log.textContent += "Headers\n";
+  capabilities.headerExtensions.forEach((header) => {
+    log.textContent += ` uri: ${header.uri}\n`;
+  });
+
+  log.textContent += "\nCodecs\n";
+  capabilities.codecs.forEach((codec) => {
+    log.textContent += ` mime type: ${codec.mimeType}\n`;
+    log.textContent += `   channels: ${codec.channels}\n`; //max channels - e.g. 2 is stereo
```
Comment (teoli2003) [inline files/en-us/web/api/rtcrtpsender/getcapabilities/index.md:187]:
```suggestion
    log.textContent += `   channels: ${codec.channels}\n`; // max channels - e.g. 2 is stereo
```

File: files/en-us/web/api/rtcrtpsender/getcapabilities/index.md [hunk 9]
```diff
@@ -81,6 +146,55 @@ function canSendH264() {
...
+    log.textContent += `   clockRate: ${codec.clockRate}\n`; // clock rate in Hz
+    log.textContent += `   sdpFmtpLine: ${codec.sdpFmtpLine}\n`; // mime meda type and subtype
+  });
+}
+```
+
+#### Result
+
+{{ EmbedLiveSample('Get all capabilities', '100%', '500px') }}
```
Comment (teoli2003) [inline files/en-us/web/api/rtcrtpsender/getcapabilities/index.md:196]:
```suggestion
{{ EmbedLiveSample('Getting all capabilities', '100%', '500px') }}
```

File: files/en-us/web/api/rtcrtpreceiver/getcapabilities/index.md [hunk 10]
```diff
@@ -81,6 +144,59 @@ function canReceiveH264() {
 }
 ```
 
+### Get all capabilities
```
Comment (teoli2003) [inline files/en-us/web/api/rtcrtpreceiver/getcapabilities/index.md:147]:
```suggestion
### Getting all capabilities
```

File: files/en-us/web/api/rtcrtpreceiver/getcapabilities/index.md [hunk 11]
```diff
@@ -44,29 +71,65 @@ As a static function, this is always called using the form:
...
+```js
+log.textContent = `RTCRtpReceiver.getCapabilities() supported: ${Object.hasOwn(
+  RTCRtpReceiver,
+  "getCapabilities"
+)}`;
+```
+
+{{ EmbedLiveSample('Feature support', '100%', '30px') }}
+
+### Check support for a particular codec
```
Comment (teoli2003) [inline files/en-us/web/api/rtcrtpreceiver/getcapabilities/index.md:128]:
```suggestion
### Checking support for a particular codec
```

File: files/en-us/web/api/rtcrtptransceiver/setcodecpreferences/index.md [hunk 12]
```diff
@@ -52,30 +55,22 @@ None ({{jsxref("undefined")}}).
 ### Exceptions
 
 - `InvalidAccessError`
```
Comment (teoli2003) [inline files/en-us/web/api/rtcrtptransceiver/setcodecpreferences/index.md:57]:
```suggestion
- `InvalidAccessError` {{domxref("DOMException")}}
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
A few minor fixes.

**Comment by hamishwillee** [comment]
Thanks very much @teoli2003 - I accepted all your review comments without changes.

**Review by teoli2003 — APPROVED** [review]
👍 The last flaw being an old one, let's merge this.

**Comment by hamishwillee** [comment]
>  The last flaw being an old one, let's merge this.

Thanks. Yes, I did check that. But it is unrelated, and fixing it would be opening yet another can of worms.

**Pull request merged**